web/src/containers/activity/Description.js

```diff
@@ -63,8 +64,8 @@ Description.propTypes = {
   updateActivity: PropTypes.func.isRequired
 };
 
-const mapStateToProps = ({ activities: { byKey } }, { aKey }) => ({
-  activity: byKey[aKey]
+const mapStateToProps = (state, props) => ({
+  activity: selectActivityByKey(state, props)
```

Comment:
**Minor tech debt:** Eventually it'd be nice to have selectors be responsible for extracting individual pieces of data from the state, like this one does.  That way the components don't have to know about the shape of the state data.